feat: add authentication documentation for REANA JupyterLab extension

tomondre opened the pull request:
The main updates provide detailed explanations of how authentication tokens are managed, how users can verify or troubleshoot authentication, and the differences between the UI extension and CLI workflows.

tomondre committed:
feat: add authentication documentation for REANA JupyterLab extension

Soap2G commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+
+## Token Reuse
+
+The REANA JupyterLab extension automatically reuses your existing VRE-issued JWT access token - the same one that authenticated your current JupyterLab session. **No additional login or authentication steps are required** to start using REANA through the extension.
```
```suggestion
The REANA JupyterLab extension automatically reuses the existing JWT access token issued by the IdP, corresponding to the one that authenticated the current JupyterLab session. **No additional login or authentication steps are required** to start using REANA through the extension.
```

tomondre replied:
I was unsure how to mention that the token is VRE requested. The VRE initiates the token request itself, hence I have mentioned like this, but I can see that it is incorrect. I will change it to your suggestion

Soap2G replied:
Thanks! "VRE-issued" it's definitely NOT incorrect (in a sense it is true), but in the perspective of having this documentation usable by a broader audience than the VRE one, I think it's better to generalise the fact that it doesn't matter the token provenance, the only important thing is that it's a valid token :)

Soap2G commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+
+## Optional Connection Configuration
+
+You can manually configure the connection to a different REANA instance by:
```
```suggestion
You can still manually configure the connection to a different REANA instance by:
```

Soap2G commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+
+1. Opening the REANA sidebar tab
+2. Clicking on the connection settings icon
+3. Entering a custom server URL and/or access token
```
```suggestion
3. Entering a custom server URL and access token
```

Soap2G commented on docs/tech-docs/services/computing-resources.md:
```diff
@@ -65,3 +65,33 @@ kubectl exec -i -t deployment/reana-server -n reana -- flask reana-admin token-g
+No refresh token is stored; only the short‑lived access token is passed.
+
+### Token storage (UI extension vs CLI)
+UI / JupyterLab extension: 
```
```suggestion
The UI / JupyterLab extension: 
```

Soap2G commented on docs/reana.md:
````diff
@@ -5,6 +5,28 @@ Explore Reana on the software's [official documentation](https://docs.reana.io/)
+Waiting for authorization...
+```
+
+> **Important:** When using the `reana-client auth` command, please note that the VRE JupyterLab extension may not be automatically updated with your new authentication token. If you're using both the command-line client and the JupyterLab extension, you may need to restart your JupyterLab session for the extension to recognize your new authentication status.
````
```suggestion
:::warning[Important] 
When using the `reana-client auth` command, please note that the VRE JupyterLab extension may not be automatically updated with your new authentication token. If you're using both the command-line client and the JupyterLab extension, you may need to restart your JupyterLab session for the extension to recognize your new authentication status.
:::
```

Soap2G replied:
Also, please have a look at https://github.com/vre-hub/vre-hub.github.io/actions/runs/16969223016/job/48101346774?pr=35

Just ping if you need any Docusaurus related help

Soap2G reviewed (commented):
Hi @tomondre, good progress!

A few minor comments below.

tomondre committed:
feat: update authentication documentation

tomondre committed:
fix: pin @docusaurus/theme-live-codeblock dependency to version 3.8.0

This change ensures all packages are the same version to avoid dependency conflicts.

garciagenrique commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+2. Checking if you can see your workflows listed
+3. Opening a workflow detail panel to see its contents
+
+If these actions succeed, your authentication is working correctly. If you see "unauthorized" errors or empty lists where you expect content, there may be an authentication issue.
```
Is the `unauthorized` feature developed and implemented ?

tomondre replied:
The "unauthorized" message is my assumption for an error message that could appear if something goes wrong. 
This PR contains documentation about multiple things that have not been developed and implemented, and I had to assume and think through some of the issues/challenges that may arise. Do you think it could be useful to make the docs more abstract and this paragraph to avoid specific values?

garciagenrique commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+
+## Token Reuse
+
+The REANA JupyterLab extension automatically reuses the existing JWT access token issued by the IdP, corresponding to the one that authenticated the current JupyterLab session. **No additional login or authentication steps are required** to start using REANA through the extension.
```
add something like "...the current JupyterLab session - in the case of the VRE, the ESCAPE INDIGO IAM"

garciagenrique commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+
+You can still manually configure the connection to a different REANA instance by:
+
+1. Opening the REANA sidebar tab
```
1. Open Reana JupyterLab by clicking on the REANA icon in the JupyterLab sidebar.

garciagenrique commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+You can still manually configure the connection to a different REANA instance by:
+
+1. Opening the REANA sidebar tab
+2. Clicking on the connection settings icon
```
Click

garciagenrique commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+
+1. Opening the REANA sidebar tab
+2. Clicking on the connection settings icon
+3. Entering a custom server URL and access token
```
Enter a REANA server URL and the REANA access token.

garciagenrique commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+2. Clicking on the connection settings icon
+3. Entering a custom server URL and access token
+
+This is useful if you need to connect to a different REANA instance than the default one configured for VRE.
```
This sentence doesn't add any relevant information. 

Please change it with the last paragraph of the page you removed:

`
After connecting to the REANA server, a notification will appear in the bottom right corner of the screen indicating that the connection was successful. You can now interact with the REANA server from within JupyterLab.
`

garciagenrique commented on docs/extensions/reana-jupyterlab/usage/authentication.md:
```diff
@@ -0,0 +1,39 @@
+
+This is useful if you need to connect to a different REANA instance than the default one configured for VRE.
+
+## Authentication Flow
```
I would move this section upper, so that we get the following order
 - Verifying Access
 - Authentication flow
 - Troubleshooting 
 - Optional Connection Configuration

garciagenrique commented on docs/reana.md:
````diff
@@ -5,6 +5,30 @@ Explore Reana on the software's [official documentation](https://docs.reana.io/)
+```text
+$ reana-client auth
+Go to: https://iam-escape.cloud.cnaf.infn.it/device
+Enter code: 4HAL4G
````
Change `4HAL4G` with ********, also below.

garciagenrique commented on docs/reana.md:
```diff
@@ -5,6 +5,30 @@ Explore Reana on the software's [official documentation](https://docs.reana.io/)
 
+# Authentication with reana-client
+
+The REANA client provides a command-line interface for authenticating with the REANA server. You can use the `reana-client auth` command which initiates an OAuth 2.0 device flow authentication:
```
Please add a link for oauth2.0 so that people can educate themselves, for example
https://auth0.com/intro-to-iam/what-is-oauth-2
or https://oauth.net/2/ 
(I prefer first link, more user-friendly. Thoughts @Soap2G)

garciagenrique commented on docs/tech-docs/services/computing-resources.md:
```diff
@@ -65,3 +65,33 @@ kubectl exec -i -t deployment/reana-server -n reana -- flask reana-admin token-g
 
+## JupyterLab REANA Extension Authentication
+
+This section explains how the VRE-provided JWT used by the REANA JupyterLab extension is injected, accessed and refreshed (or not) during a user session.
```
user friendly explanatory link for JWT too, please.

garciagenrique commented on docs/tech-docs/services/computing-resources.md:
```diff
@@ -65,3 +65,33 @@ kubectl exec -i -t deployment/reana-server -n reana -- flask reana-admin token-g
+No refresh token is stored; only the short‑lived access token is passed.
+
+### Token storage (UI extension vs CLI)
+The UI / JupyterLab extension: 
```
By UI you mean the Jupyterlba extension, right ? In this case remove UI to avoid misunderstandings, please

garciagenrique commented on docs/tech-docs/services/computing-resources.md:
```diff
@@ -65,3 +65,33 @@ kubectl exec -i -t deployment/reana-server -n reana -- flask reana-admin token-g
+- Uses this token for all API calls to the REANA server.
+
+CLI (`reana-client`):
+- Loads the token from the REANA config file written at spawn (or later replaced by running `reana-client auth`, which performs an OAuth 2.0 device flow and rewrites the stored token).
```
change it with `written "during the spawning of the user session"`

garciagenrique commented on docs/tech-docs/services/computing-resources.md:
```diff
@@ -65,3 +65,33 @@ kubectl exec -i -t deployment/reana-server -n reana -- flask reana-admin token-g
+During user pod creation the VRE spawner:
+- Obtains (or reuses) the user's already validated access token.
+- Injects it as an environment variable (`REANA_ACCESS_TOKEN`).
+- Writes a lightweight config file with the token for `reana-client` CLI
```
Add a point at the end of the sentence, please

garciagenrique reviewed (changes requested):
Small comments, otherwise looks fine to me ! Good job @tomondre !

tomondre committed:
feat: enhance authentication documentation for REANA JupyterLab extension
